centreon/www/include/configuration/configObject/host/DB-Func.php

```diff
@@ -3121,105 +3118,74 @@ static function (int $key, string $name, string $value) use ($formData): array {
-                : null,
-            'is_activated' => (bool) ($formData['host_activate']['host_activate'] ?: false),
-            'templates' => array_map(static fn(string $id): int => (int) $id, $formData['tpSelect'] ?? []),
-            'categories' => array_map(static fn(string $id): int => (int) $id, $formData['host_hcs'] ?? []),
-            'groups' => array_map(static fn(string $id): int => (int) $id, $formData['host_hgs'] ?? []),
-            'macros' => array_map(
-                static function (int|string $key, string $name, string $value) use ($formData): array {
-                    return [
-                        'name' => $name,
-                        'value' => $value,
-                        'is_password' => (bool) ($formData['macroPassword'][$key] ?? false),
-                        'description' => $formData["macroDescription_{$key}"],
-                    ];
-                },
-                array_keys($formData['macroInput'] ?? []),
-                $formData['macroInput'] ?? [],
-                $formData['macroValue'] ?? []
-            ),
-            'event_handler_enabled' => isset($formData['host_event_handler_enabled']['host_event_handler_enabled'])
-                ? (int) $formData['host_event_handler_enabled']['host_event_handler_enabled']
-                : null,
-            'event_handler_command_id' => isset($formData['command_command_id2']) && '' !== $formData['command_command_id2']
-                ? (int) $formData['command_command_id2']
-                : null,
-        ];
-    } else {
-        return [
-            'name' => $formData['host_name'],
-            'address' => $formData['host_address'],
-            'monitoring_server_id' => (int) $formData['nagios_server_id'] ?: null,
-            'alias' => $formData['host_alias'] ?: null,
-            'snmp_version' => $formData['host_snmp_version'] ?: null,
-            'snmp_community' => $formData['host_snmp_community'] ?: null,
-            'note_url' => $formData['ehi_notes_url'] ?: null,
-            'note' => $formData['ehi_notes'] ?: null,
-            'action_url' => $formData['ehi_action_url'] ?: null,
-            'icon_id' => '' !== $formData['ehi_icon_image']
-                ? (int) $formData['ehi_icon_image']
-                : null,
+    $payload = [
+        'name' => $formData['host_name'],
+        'address' => $formData['host_address'],
+        'monitoring_server_id' => (int) $formData['nagios_server_id'] ?: null,
+        'alias' => $formData['host_alias'] ?: null,
+        'snmp_version' => $formData['host_snmp_version'] ?: null,
+        'snmp_community' => $formData['host_snmp_community'] ?: null,
+        'note_url' => $formData['ehi_notes_url'] ?: null,
+        'note' => $formData['ehi_notes'] ?: null,
+        'action_url' => $formData['ehi_action_url'] ?: null,
+        'icon_id' => '' !== $formData['ehi_icon_image']
+            ? (int) $formData['ehi_icon_image']
```

Comment:
Casting a string as an integer can lead to unexpected behavior
`(int) "hello"` => 0 (In our database system, all identifiers are > 0)

Reply:
It's for the API payload, the validation of the payload will handled this kind of error